### DISCUSSION TITLE
Question about non-core functionality

### DISCUSSION THREAD

**Discussion by willcohen** [dcoles/flipperzero-rs#40] [Q&A] [answered]
I'm looking into trying to build some additional IR functionality to a F0 (so it can send/receive via [kermit](https://www.kermitproject.org/ek.html)). It's extremely possible it'll be a dead end due to hardware constraints, but if I were to get it working -- probably using bindgen around e-kermit -- would this be the kind of thing that you'd prefer live in its own separate library, or that might end up as a PR here as a separate crate? Nothing about this is mission-critical for the majority of flipper users, but I'm not sure if you'd like this repo to be a clearinghouse for various rust-related community efforts or if you'd like to keep it to the basics that apply to everyone once they get filled out?

**Comment by dcoles** [comment, marked as answer]
Hi @willcohen,

That's a really neat idea. I might have to dig up an old IR-blaster to the IR functionality out.

I probably wouldn't accept it in this repository (wrappers for the official Flipper Zero SDK) because I want to control the scope of these crates so that the maintenance burden doesn't become unmanageable. So I would suggest creating a dedicated repository for a kermit file-transfer app.

It's quite possible the [flipperzero-firmware](https://github.com/flipperdevices/flipperzero-firmware) project might be interested in supporting file and generic data transfers over unreliable serial (IR, UART, Bluetooth, ...), of which kermit would be a excellent fit. This also solves the issue of building a C-library like e-kermit (currently the `flipperzero` crates don't depend on a C compiler/toolchain).

**Reply by willcohen** [reply]
Makes sense. Many thanks! Will keep poking. Also -- many thanks for this work, it's much appreciated.
